make.mk

```diff
@@ -242,3 +242,37 @@ endif
 ifdef CLOCK_FACE_24H_ONLY
 CFLAGS += -DCLOCK_FACE_24H_ONLY
 endif
+
+# DATE = X
+#  YEAR = Sets the year and timezone to the PC's
+#  DAY = Sets the default time down to the day (year, month, day, timezone)
+#  MIN = Sets the default time down to the minute (year, month, day, timezone, hour, minute)
+ifdef DATE
+TIMEZONE := $(shell date +%z | awk '{print substr($$0, 1, 3) * 60 + substr($$0, 4, 2)}')
+CURRENT_YEAR := $(shell echo $$(($(shell date +"%Y") - 2020)))
+CURRENT_MONTH := $(shell date +"%-m")
+CURRENT_DAY := $(shell date +"%-d")
+CURRENT_HOUR := $(shell date +"%-H")
+CURRENT_MINUTE := $(shell date +"%-M")
```

Comment:
This adds a direct dependency on GNU coreutils for building the firmware. I consider that to be perfectly OK but then again I've been running Linux for over a decade. I have no idea how this impacts Windows users. Maintainers might want to double check this.